x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.core.ml.dataframe.DataFrameAnalyticsConfigTests.randomValidId;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+public class DataFrameAnalyticsConfigUpdateTests extends AbstractSerializingTestCase<DataFrameAnalyticsConfigUpdate> {
+
+    @Override
+    protected DataFrameAnalyticsConfigUpdate doParseInstance(XContentParser parser) throws IOException {
+        return DataFrameAnalyticsConfigUpdate.PARSER.apply(parser, null).build();
+    }
+
+    @Override
+    protected DataFrameAnalyticsConfigUpdate createTestInstance() {
+        return randomUpdate(randomValidId());
+    }
+
+    @Override
+    protected Writeable.Reader<DataFrameAnalyticsConfigUpdate> instanceReader() {
+        return DataFrameAnalyticsConfigUpdate::new;
+    }
+
+    public static DataFrameAnalyticsConfigUpdate randomUpdate(String id) {
+        DataFrameAnalyticsConfigUpdate.Builder builder = new DataFrameAnalyticsConfigUpdate.Builder(id);
+        if (randomBoolean()) {
+            builder.setDescription(randomAlphaOfLength(20));
+        }
+        if (randomBoolean()) {
+            builder.setModelMemoryLimit(new ByteSizeValue(randomNonNegativeLong()));
+        }
+        if (randomBoolean()) {
+            builder.setAllowLazyStart(randomBoolean());
+        }
+        return builder.build();
+    }
+
+    public void testMergeWithConfig() {
+        String id = randomValidId();
+        {
```

Comment:
I'd rather we split this in 4 tests with names for each property that is updated. The reason I think that's beneficial, is that if the test fails, the test title immediately tells us what went wrong.

Reply:
Done.